src/server/heartbeat.js

```diff
@@ -4,7 +4,8 @@ const mongodb = require('mongodb');
 const MongoClient = mongodb.MongoClient;
 
 const db = require('../../.environment.json');
-const url = 'mongodb://' + db.database.url + ':' + db.database.port + '/' + db.database.database;
+const auth = process.env.DB_USER ? process.env.DB_USER + ':' + process.env.DB_PASS + '@' : '';
+const url = 'mongodb://' + auth + process.env.DB_HOST + ':' + process.env.DB_PORT + '/' + process.env.DB_NAME;
```

Comment:
does a url like this `mongodb://:localhost:27017/black-pearl` work?

also definitely put this part into a function or a separate file